### PR TITLE
Update SITL information

### PR DESCRIPTION
**Pull request by Hwurzburg**
needs a careful read, extensively re-organized into evoking the sims by sim_vehicle or in Missionplanner, separating params that control the sim to a new page, and a page on how to add simulated devices

also update the Mission Planner SIM page with new command line options that are useful

### CODE REVIEW

File: dev/source/docs/SITL_simulation_parameters.rst [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+   When you change a parameter the value remains in the virtual EEPROM
+   after you restart SITL. Remember to change it back if you don't want it
+   any more (or :ref:`reload/reset the parameters <using-sitl-for-ardupilot-testing_loading_a_parameter_set>`). 
+
+
+Changing the speed of the simulation
+------------------------------------
+
+Most of the simulator backends (native, RealFlight,etc) support changing the speed while
+running (NOT Realflight). Just set the ``SIM_SPEEDUP`` parameter as needed. A value of 1
```
Comment (peterbarker) [inline dev/source/docs/SITL_simulation_parameters.rst:39]: Better make up your mind on RealFlight :-)

File: dev/source/docs/SITL_simulation_parameters.rst [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+Loading a parameter set
+-----------------------
+
+When starting SITL the first time, the device will be configured with
+default parameters for the vehicle/frame type. Typically, you will want to replace these with
+values that simulate the desired vehicle and environment conditions.
+
+::
+
+    param load <<path to parameter file>
```
Comment (peterbarker) [inline dev/source/docs/SITL_simulation_parameters.rst:54]:
```suggestion
    param load <path to parameter file>
```

File: dev/source/docs/SITL_simulation_parameters.rst [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+``-w`` flag.
+
+.. note:: You may have to load them again, after typing ``param fetch``, in order to load parameters that require enabling before presenting their parameter set.  And, in some cases, you may even need to restart SITL in order for some new parameters, such as output function changes, to take effect. This can be avoided if you add the parameter file during the start of SITL with the "--add-param-file=*pathtofile* "....ie: sim_vehicle.py -f flightaxis:192.168.x.x - -map - -console --add-param-file=*pathtofile* -w. This adds the param file as a default and then wipes any previous param changes that may exist in the simulation directory. 
+
+Parameters can also be saved. For example, to save the parameters into
+the present working directory you might do:
+
+::
+
+    param save ./myparams.parm
```
Comment (peterbarker) [inline dev/source/docs/SITL_simulation_parameters.rst:69]: Perhaps mention `param help` for other options?

File: dev/source/docs/SITL_simulation_parameters.rst [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+
+.. note:: the "default" parameter set emulates the parameter default values contained in the firmware. These used at startup for parameters, UNLESS the user has previously changed them for a parameter(s), in which case those changed values are used. In SITL, these changes are stored in a file in the simulation startup directory named ``eeprom.bin``. The above commands loads user set values for the parameters, not the default values created at simulation start.
```
Comment (peterbarker) [inline dev/source/docs/SITL_simulation_parameters.rst:71]:
```suggestion
.. note:: the "default" parameter set emulates the parameter default values contained in the firmware. These are used at startup for parameters, UNLESS the user has previously changed them for a parameter(s), in which case those changed values are used. In SITL, these changes are stored in a file in the simulation startup directory named ``eeprom.bin``. The above commands loads user set values for the parameters, not the default values created at simulation start.
```

I think this should be rewritten - I'm having trouble following it...

File: dev/source/docs/SITL_simulation_parameters.rst [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+
+    param set SIM_RC_FAIL 1
+
+This simulates the complete loss of RC input. If you just want to
+simulate low throttle (below throttle failsafe level) then you can do
+that with the RC command:
+
+::
+
+    rc 3 900
```
Comment (peterbarker) [inline dev/source/docs/SITL_simulation_parameters.rst:89]: There's actually a `SIM_RC_FAIL` value of 2 which is used to simulate receiver-goes-to-bind-time-values.

File: dev/source/docs/adding_simulated_devices.rst [hunk 6]
```diff
@@ -0,0 +1,319 @@
...
+
+Adding a Gimbal
+===============
+
+SITL can simulate a virtual gimbal.
+
+.. note::
+
+   Gimbal simulation causes SITL to start sending
+   `MOUNT_STATUS <https://mavlink.io/en/messages/ardupilotmega.html#MOUNT_STATUS>`__
```
Comment (peterbarker) [inline dev/source/docs/adding_simulated_devices.rst:29]: This is no longer true; we send a `GIMBAL_DEVICE_ATTITUDE_STATUS` message.  And it only has a quaternion in it, so good luck! :-)

( @rmackay9 )

File: dev/source/docs/adding_simulated_devices.rst [hunk 7]
```diff
@@ -0,0 +1,319 @@
...
+
+::
+
+   ls -lt logs
+
+Choose the youngest, then:
+
+::
+
+   mavlogdump --type PL logs/<youngest>
```
Comment (peterbarker) [inline dev/source/docs/adding_simulated_devices.rst:225]:
```suggestion
   mavlogdump.py --type PL logs/<youngest>
```

**Review by peterbarker — CHANGES_REQUESTED** [review]
I wonder if we should mention the comments I've been dropping into the simulated devices to aid in use them.  

e.g. https://github.com/ardupilot/ardupilot/blob/master/libraries/SITL/SIM_RF_Lanbao.h#L18

File: dev/source/docs/SITL_simulation_parameters.rst (hunk 1 above)
Comment (Hwurzburg) [reply dev/source/docs/SITL_simulation_parameters.rst:39]: fixed

File: dev/source/docs/SITL_simulation_parameters.rst (hunk 2 above)
Comment (Hwurzburg) [reply dev/source/docs/SITL_simulation_parameters.rst:54]: existing...fixed

File: dev/source/docs/SITL_simulation_parameters.rst (hunk 3 above)
Comment (Hwurzburg) [reply dev/source/docs/SITL_simulation_parameters.rst:69]: added

File: dev/source/docs/SITL_simulation_parameters.rst (hunk 5 above)
Comment (Hwurzburg) [reply dev/source/docs/SITL_simulation_parameters.rst:89]: added....actually I misunderstood this when updating the RC_FAIL param....will fix that now

File: dev/source/docs/adding_simulated_devices.rst (hunk 6 above)
Comment (Hwurzburg) [reply dev/source/docs/adding_simulated_devices.rst:29]: can fix later once Randy updates

File: dev/source/docs/adding_simulated_devices.rst (hunk 7 above)
Comment (Hwurzburg) [reply dev/source/docs/adding_simulated_devices.rst:225]: fixed

**Comment by Hwurzburg** [comment]
> I wonder if we should mention the comments I've been dropping into the simulated devices to aid in use them.
> 
> e.g. https://github.com/ardupilot/ardupilot/blob/master/libraries/SITL/SIM_RF_Lanbao.h#L18

so are the rangefinder instructions incorrect? or just incomplete for some RF? as to mentions, not sure how to definitively instruct someone since we have a mix of places the added devices exist, right?

File: dev/source/docs/SITL_simulation_parameters.rst (hunk 4 above)
Comment (Hwurzburg) [reply dev/source/docs/SITL_simulation_parameters.rst]: done

**Comment by peterbarker** [comment]
> 

The rangefinder instructions use the in-built analogue rangefinder, but we simulate pretty much every rangefinder we support - so we get testing of the protocol parsers.  I don't think we simulate their maximum ranges, however.

**Pull request merged**